lib/PAUSE/Indexer/Errors.pm

```diff
@@ -124,30 +141,48 @@ EOF
 };
 
 pkg_error dual_newer => {
-  # TODO bring back parameters
   header  => 'Not indexed because of an newer dual-life module',
-  body    => <<'EOF',
-Not indexed because package $opack in file $ofile has a dual life in $odist.
-The other version is at $oldversion, so not indexing seems okay.
+  body    => sub {
+    my ($old) = @_;
+
+    _assert_args_present(db_conflict => $old, [ qw(package file dist version) ]);
+
+    return <<"EOF";
+Not indexed because package $old->{pack} in file $old->{file} has a dual life
+in $old->{dist}.  The other version is at $old->{version}, so not indexing
+seems okay.
 EOF
+  },
 };
 
 pkg_error dual_older => {
-  # TODO bring back parameters
   header  => 'Not indexed because of an older dual-life module',
-  body    => <<'EOF',
-Not indexed because package $opack in file $ofile seems to have a dual life in
-$odist. Although the other package is at version [$oldversion], the indexer
-lets the other dist continue to be the reference version, shadowing the one in
-the core.  Maybe harmless, maybe needs resolving.
+  body    => sub {
+    my ($old) = @_;
+
+    _assert_args_present(db_conflict => $old, [ qw(package file dist version) ]);
+
+    return <<"EOF";
+Not indexed because package $old->{pack} in file $old->{file} seems to have a
```

Comment:
You expect 'package' but you're using 'pack' so this throws a warning and doesn't fill in